File: examples/java/spring-boot/one_consumer_two_providers/account-provider/src/main/java/br/com/zup/pact/accountprovider/dto/AccountDetailsDTO.java
package br.com.zup.pact.accountprovider.dto;

import br.com.zup.pact.accountprovider.entity.Account;
import br.com.zup.pact.accountprovider.enums.AccountType;
import java.math.BigDecimal;
import java.util.Objects;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class AccountDetailsDTO {

    private Integer accountId;
    private BigDecimal balance;
    private AccountType accountType;


    public static Account fromDtoToEntity(AccountDetailsDTO accountDetailsDTO) {
        if (Objects.nonNull(accountDetailsDTO)) {
            return Account.builder()
                    .accountType(accountDetailsDTO.getAccountType())
                    .balance(accountDetailsDTO.getBalance())
                    .build();

        }
        return null;
    }
}

File: example/java/kotlin/provider/src/main/kotlin/br/com/zup/pact/provider/dto/AccountDetailsDTO.kt
package br.com.zup.pact.provider.dto

import br.com.zup.pact.provider.enums.AccountType
import br.com.zup.pact.provider.resource.AccountDetailsResponse
import br.com.zup.pact.provider.resource.AccountDetailsResponseOrBuilder

data class AccountDetailsDTO(
        val accountId: Int,
        val balance: Double,
        val accountType: AccountType
) {
//
//    fun toAccountDetailsResponse(): AccountDetailsResponse {
//        return AccountDetailsResponse.newBuilder()
//                .setAccountId(accountId.toString())
//                .setBalance(balance.toString())
//                .accountType(accountType.toString())
//                .build()
//    }
}

